metrifier/config.go

```diff
@@ -0,0 +1,89 @@
+package metrifier
+
+import (
+	"time"
+
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+// Config is used to configure a Metrifier.
+type Config struct {
+	// System identifies the application being instrumented.
+	//
+	// This field is mandatory.
+	//
+	// If you have multiple deployments consider adding the environment here.
+	// Examples: myserviceprod, myservicetesting
+	System string
+
+	// Subsystem identifies which part of the code is being instrumented.
+	//
+	// This field is mandatory.
+	//
+	// Examples: myserviceendpoint, myentityrepository, myjobqueue
+	Subsystem string
+
+	// ExtraLabels are extra prometheus labels that will be attached to the metric.
+	//
+	// The Metrifier will always create and manage the "error_code" label
+	// automatically. The ExtraLabels field is for any extra label that
+	// will be (and must be) passed by calling the Span.WithLabels.
+	ExtraLabels []string
+
+	// The bellow fields are passed directly to promethues.
+
+	// ConstLabels are used to attach fixed labels to this metric. Metrics
+	// with the same fully-qualified name must have the same label names in
+	// their ConstLabels.
+	//
+	// Due to the way a Summary is represented in the Prometheus text format
+	// and how it is handled by the Prometheus server internally, “quantile”
+	// is an illegal label name. Construction of a Summary or SummaryVec
+	// will panic if this label name is used in ConstLabels.
+	//
+	// ConstLabels are only used rarely. In particular, do not use them to
+	// attach the same labels to all your metrics. Those use cases are
+	// better covered by target labels set by the scraping Prometheus
+	// server, or by one specific metric (e.g. a build_info or a
+	// machine_role metric). See also
+	// https://prometheus.io/docs/instrumenting/writing_exporters/#target-labels,-not-static-scraped-labels
+	ConstLabels stdprometheus.Labels
+
+	// The bellow fields applies only to summary
```

Comment:
Nit: by grouping in a inner struct, the grouping might become clearer than with a label comment